### DISCUSSION TITLE
Whats the difference between input prefix / suffix (inp_pfx/inp_sfx) and line  prefix / suffix (line_pfx/line_sfx)?

### DISCUSSION THREAD

**Discussion by danemadsen** [ggml-org/llama.cpp#3691] [Q&A] [answered]
I'm looking to integrate prompt templates directly into code for my app and I would like to know what each of these variables mean? They both seem to have the same purpose so why have both?

**Comment by KerfuffleV2** [comment, marked as answer]
Trying to understand the logic in `main`? Good luck!

`inp_pfx` and `inp_sfx` are (as far as I know) only used when instruct mode is turned on.

https://github.com/ggerganov/llama.cpp/blob/465219b9143ac01db0990bbcb0a081ef72ec2008/examples/main/main.cpp#L321-L323

The values are hardcoded to the old Alpaca format.

On the other hand, `line_prefix` and `line_suffix` seem to come from the user specified parameters (on the commandline `--in-prefix`, `--in-suffix`)

https://github.com/ggerganov/llama.cpp/blob/465219b9143ac01db0990bbcb0a081ef72ec2008/examples/main/main.cpp#L765-L768

If that's not enough information to do what you want then you might also try looking at the log file and even adding a few of your own `LOG(...)` lines if needed.
